compiler/parser/RdrHsSyn.hs

```diff
@@ -2868,35 +2875,56 @@ mkChoiceInstanceDecl templateName CombinedChoiceData { ccdChoiceData = ChoiceDat
 
 mkInterfaceFixedChoiceInstanceDecl :: Located RdrName -> InterfaceChoiceSignature -> [LHsDecl GhcPs]
 mkInterfaceFixedChoiceInstanceDecl tycon InterfaceChoiceSignature {..} =
-  [ mkInstance "HasToAnyChoice" simpleContext (mkPrimMethod "_toAnyChoice" "EToAnyChoice")
-  , mkInstance "HasFromAnyChoice" simpleContext (mkPrimMethod "_fromAnyChoice" "EFromAnyChoice")
-  , mkInstance "HasExercise" exerciseContext (mkTemplateClassMethod "exercise" [cid, arg]
-      (mkPrimitive "primitive" "UExerciseInterface"
-        `mkApp` (mkParExpr $ mkQualVar (mkVarOcc "toInterfaceContractId")
-                  `mkAppType` rdrNameToType tycon
-                  `mkApp` mkUnqualVar (mkVarOcc "cid"))
-        `mkApp` (mkUnqualVar (mkVarOcc "arg"))
-        `mkApp` (mkParExpr $ mkQualVar (mkVarOcc "_typeRepForInterfaceExercise")
-                  `mkApp` mkUnqualVar (mkVarOcc "cid")))
-      Nothing)
+  [ mkInstance "HasToAnyChoice"
+      [paramType, choiceType, returnType]
+      simpleContext
+      (mkPrimMethod "_toAnyChoice" "EToAnyChoice")
+  , mkInstance "HasFromAnyChoice"
+      [paramType, choiceType, returnType]
+      simpleContext
+      (mkPrimMethod "_fromAnyChoice" "EFromAnyChoice")
+  , mkInstance "HasExerciseInterface"
```

Comment:
this is pretty much the same as we generated before for `HasExercise`, except adding the class argument for the interface, and the method argument for the predicate